Automattic/pocket-casts-ios
# Serial show support

| 📘 Part of: [Improve Support for Serial Shows](https://linear.app/a8c/project/improve-support-for-serial-shows-fba2e2f0f9fc/overview)
|:---:|

Fixes [PCIOS-2](https://linear.app/a8c/issue/PCIOS-2/implement-new-serial-sorting)
Fixes [PCIOS-3](https://linear.app/a8c/issue/PCIOS-3/implement-new-serial-seasons-actions)
Fixes [PCIOS-4](https://linear.app/a8c/issue/PCIOS-4/implement-new-analytics-for-season-actions)


This PR adds support for improved fiction shows, AKA serial shows, that have season and episode information. For example "White Vault".
The changes done are inspired in this blog post: https://www.theend.fyi/article/unlocking-the-potential-of-fiction-podcasts-a-call-to-podcast-app-developers

It adds the following features:

- A new sort type, `Serial`, that orders the episode by ascending season number and episode number
- Episodes that don't have a Season and Episode number are group in section called Extras in the end of the list of episodes
- By default if a podcast is added and it has the `showType == serial` the default sort order for the episodes is set to Serial and Grouping by Season
- Adds a new button to support archive, select and download of all episodes of a season.

https://github.com/user-attachments/assets/73256146-bc92-47a6-a264-9f4e3ab55db2

## To test

1. Start the app
2. Enable the FF for tracksLogging
3. Go to discover and search for serial show that you still don't follow. Ex: White Vault ( normally podcasts in the Fiction category have this type)
4. Check that by default it shows the podcast using the serial sort order and grouped by season
5. Check that in front of each season it shows a [...] button
6. Tap on that button and check if you see options for: Select All, Download All, Select All
7. Check if you see in the console the event: `podcastScreenSeasonOptionsTapped`
8. Try each option and see if they work correctly.
9. Check for each action you see the corresponding event: `podcastScreenSeasonOptionsSelectAllTapped`, `podcastScreenSeasonOptionsDownloadAllTapped`, `podcastScreenSeasonOptionsArchiveAllTapped`
10. Follow the podcast
11. Check if the grouping and sort order are kept
12. Check the following actions:
13. Archive all the episodes of a season by tapping on more button for a season
14. Now tap on the more button of the season and check if the option Unarchive All shows
15. Tap on it and see if unarchive runs correctly
16. Download all the episodes of a season
17.  Now tap on the more button of the season and check if the option Remove All shows
18. Tap on it and check if all downloads are removed.

## Checklist
